src/lib.rs

```diff
@@ -47,21 +49,29 @@ impl From<reqwest::Error> for Error {
     }
 }
 
-pub async fn serve(addr: SocketAddr, market_url: &str) -> Result<(), Error> {
+impl From<http::uri::InvalidUri> for Error {
```

Comment:
You actually don't need this `InvalidUri`, the `http::Error` type implements `From<InvalidUri> for http::Error`, which means that using `?` will auotmaticaly use this From implementation and make the `InvalidUrl` into a `http::Error`.
hint: We already have this error in our `enum Error` :raised_hands: